src/python/pants/engine/scheduler.py

```diff
@@ -269,17 +269,17 @@ def add_root_selection(self, execution_request, subject, product):
                                                      execution_request,
                                                      self._to_key(subject),
                                                      self._to_constraint(product))
-    if res.is_throw:
-      raise self._from_value(res.value)
+    self._from_py_result(res)
 
   def visualize_to_dir(self):
     return self._native.visualize_to_dir
 
   def to_keys(self, subjects):
     return list(self._to_key(subject) for subject in subjects)
 
-  def pre_fork(self):
-    self._native.lib.scheduler_pre_fork(self._scheduler)
+  def with_fork_context(self, func):
```

Comment:
Can you add a quick pydoc explaining what this is and how it should be used?

Reply:
I'll refer to the rust docs on the topic (in `lib.rs`).